### PR TITLE
Delete failed cert after 7 days from kubernetes

### PR DESCRIPTION
**Pull request by aalvarado1**
If a certificate failed provisioning then delete the certificate object after 7 days.

### CODE REVIEW

**Review by jimbobhickville — COMMENTED** [review]
lgtm. Should probably get someone more familiar with Go to weigh in on it.

File: k8s.go [hunk 1]
```diff
@@ -126,6 +126,15 @@ func (k K8sClient) deleteSecret(namespace string, key string) error {
 	return k.c.Secrets(namespace).Delete(key, nil)
 }
 
+func (k K8sClient) deleteCertificate(c Certificate, namespace string) error {
+	log.Printf("About to delete certificate " + c.Metadata.Name + " / " + namespace)
```
Comment (ctriv) [inline k8s.go]: I would either use the printf formatting here or call `log.Print` instead here.

File: processor.go [hunk 2]
```diff
@@ -639,3 +641,14 @@ func valueOrDefault(a, b string) string {
 	}
 	return b
 }
+
+func (p *CertProcessor) deleteFailedCertIfNeeded(c Certificate, namespace string) {
+	cutoff := c.Metadata.CreationTimestamp.Time.UTC().AddDate(0, 0, 7)
+
+	if cutoff.Before(time.Now().UTC()) {
+		error := p.k8s.deleteCertificate(c, namespace)
```
Comment (ctriv) [inline processor.go]: `error` is an internal interface in go.  It's generally bad form to use the same name for a variable, that's why there's an idiom of naming error variables `err`.

Comment (aalvarado1) [reply processor.go]: Ah gotya, thanks. Will change that to err instead of error.

File: k8s.go (hunk 1 above)
Comment (aalvarado1) [reply k8s.go]: Ah yes, forgot about that one as I did use it in the other log statements. Will change.

**Pull request merged**